src/ontodia/diagram/halo/halo.ts

```diff
@@ -0,0 +1,54 @@
+import * as joint from 'jointjs';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+import { HaloMarkup } from './haloMarkup';
+
+export interface HaloOptions {
+    paper: joint.dia.Paper;
+    cellView: joint.dia.CellView;
+    onDelete: () => void;
+    onExpand: () => void;
+}
+
+export class Halo {
+    private container: HTMLElement;
+
+    constructor(private options: HaloOptions) {
+        this.render();
+
+        this.options.cellView.model.on('change:position', () => this.update());
+        this.options.cellView.model.on('change:isExpanded', () => this.update());
+        this.options.paper.on('scale', () => this.update());
+    }
+
+    private render() {
+        this.container = document.createElement('div');
+        this.container.className = 'ontodia-halo';
+        this.options.paper.el.appendChild(this.container);
+
+        this.update();
+
+        ReactDOM.render(React.createElement(HaloMarkup, {
+            cellView: this.options.cellView,
+            onDelete: this.options.onDelete,
+            onExpand: this.options.onExpand,
+        }), this.container);
+    }
+
+    private update() {
```

Comment:
Better extract `BBox` into state of React component and use `setState()` / `render()` to update `top`, `left`, `width`, `height`.
Instead you may create an instance of `Backbone.Model` then use `listenTo()` to subscribe and `stopListening()` to remove subscriptions all at once.